centrifuge.wdl

```diff
@@ -1,6 +1,109 @@
 #
 # Tasks from centrifuge
+task build {
+
+    File conversionTable
+    File taxonomyTree
+    File inputFasta
+    String centrifugeIndexBase
+    String? preCommand
+    String? centrifugeBuildExecutable = "centrifuge-build"
+    #Boolean? c = false
+    Boolean? largeIndex = false
+    Boolean? noAuto = false
+    Int? bMax
+    Int? bMaxDivn
+    Boolean? noDiffCover = false
+    Boolean? noRef = false
+    Boolean? justRef = false
+    Int? offRate
+    Int? fTabChars
+    File? nameTable
+    File? sizeTable
+    Int? seed
+    Int? threads = 1
+    Int? kmerCount
+
+    command {
+        set -e -o pipefail
+        ${preCommand}
+        ${"mkdir -p $(dirname " + centrifugeIndexBase + ")"}
+        ${centrifugeBuildExecutable} \
+        ${true='--large-index' false='' largeIndex} \
+        ${true='--noauto' false='' noAuto} \
+        ${'--bmax ' + bMax} \
+        ${'--bmaxdivn ' + bMaxDivn} \
+        ${true='--nodc' false='' noDiffCover} \
+        ${true='--noref' false='' noRef} \
+        ${true='--justref' false='' justRef} \
+        ${'--offrate ' + offRate} \
+        ${'--ftabchars ' + fTabChars} \
+        ${'--name-table ' + nameTable } \
+        ${'--size-table ' + sizeTable} \
+        ${'--seed ' + seed} \
+        ${'--kmer-count' + kmerCount} \
+        ${'--threads ' + threads} \
+        --conversion-table ${conversionTable} \
+        --taxonomy-tree ${taxonomyTree} \
+        ${inputFasta} \
+        ${centrifugeIndexBase}
+    }
+    runtime {
+        cpu: select_first([threads])
+    }
+}
+
+task classify {
+    String outputDir
+    Boolean compressOutput = true
+    String? preCommand
+    String indexPrefix
+    File? unpairedReads
+    File read1
+    File? read2
+    Boolean? fastaInput
+    String? outputFilePath = outputDir + "/centrifuge.out"
+    String? reportFilePath = outputDir + "/centrifuge_report.tsv"
+    String? metFilePath # If this is specified, the report file is empty
+    Int? assignments
+    Int? minHitLen
+    Int? minTotalLen
+    Array[String]? hostTaxIds
+    Array[String]? excludeTaxIds
+    Int? threads
+    Int? memory
+
+    command {
+        set -e -o pipefail
+        mkdir -p ${outputDir}
+        ${preCommand}
+        centrifuge \
+        ${"-p " + threads} \
+        ${"-x " + indexPrefix} \
+        ${true="-f" false="" fastaInput} \
+        ${true="-k " false="" defined(assignments)} ${assignments} \
+        ${true="-1 " false="-U " defined(read2)} ${read1} \
+        ${"-2 " + read2} \
+        ${"-U " + unpairedReads} \
+        ${"--report-file " + reportFilePath} \
+        ${"--min-hitlen " + minHitLen} \
+        ${"--min-totallen " + minTotalLen} \
+        ${true="--host-taxids " false="" defined(hostTaxIds)} ${sep=',' hostTaxIds} \
+        ${true="--exclude-taxids " false="" defined(excludeTaxIds)} ${sep=',' excludeTaxIds} \
+        ${true="| gzip -c > " false="-S " compressOutput}${outputFilePath}${true=".gz" false="" compressOutput}
```

Comment:
I would add a space between the first two `${...}`. This would make it a bit more readable, I think.

In addition, it might be better to define a variable with the outputFilePath, rather than doing it the command and then again in the output.
eg. place `String finalOutputPath = if (compressOutput) then select_first([outputFilePath + ".gz"]) else select_first([outputFilePath])` above the command.
This would make sure that you can't forget to change one of them when you change the other one, because there is only one to change.